Documentation/ceph-csi-drivers.md

```diff
@@ -0,0 +1,382 @@
+    csiProvisionerSecretName: csi-rbd-secret
+    csiProvisionerSecretNamespace: default
+    csiNodePublishSecretName: csi-rbd-secret
+    csiNodePublishSecretNamespace: default
+
+    # Ceph users for operating RBD
+    adminid: admin
+    userid: kubernetes
+    # uncomment the following to use rbd-nbd as mounter on supported nodes
+    #mounter: rbd-nbd
+reclaimPolicy: Delete
+```
+### Create the Secret:
+##### secret.yaml
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: csi-rbd-secret
+  namespace: default 
+data:
+  # Key value corresponds to a user name defined in ceph cluster
+  admin: BASE64-ENCODED-PASSWORD
+  # Key value corresponds to a user name defined in ceph cluster
+  kubernetes: BASE64-ENCODED-PASSWORD
+  # if monValueFromSecret is set to "monitors", uncomment the
+  # following and set the mon there
+  #monitors: BASE64-ENCODED-Comma-Delimited-Mons
+```
+Here, you need your ceph admin/user password encoded in base64. Run `ceph auth ls` in one of your ceph pod, encode the key of your admin/user and replace `BASE64-ENCODED-PASSWORD` by your encoded key.
+### Create the PersistentVolumeClaim:
+##### pvc.yaml
+```
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: rbd-pvc
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+  storageClassName: csi-rbd
+```
+Make sure your `storageClassName` is the name of the StorageClass previously defined in storageclass.yaml
+### Create the demo Pod:
+##### pod.yaml
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: csirbd-demo-pod
+spec:
+  containers:
+   - name: web-server
+     image: nginx 
+     volumeMounts:
+       - name: mypvc
+         mountPath: /var/lib/www/html
+  volumes:
+   - name: mypvc
+     persistentVolumeClaim:
+       claimName: rbd-pvc
+       readOnly: false
+```
+### Verify the PVC has successfully been created:
+```
+# kubectl get pvc
+NAME      STATUS   VOLUME                 CAPACITY   ACCESS MODES   STORAGECLASS   AGE
+rbd-pvc   Bound    pvc-c20495c0d5de11e8   1Gi        RWO            csi-rbd        21s
+```
+If your PVC status isn't `Bound`, check the csi-rbdplugin logs to see what's preventing the PVC from being up and bound.
+
+When running `rbd list block --pool [yourpool]` in one of your ceph pod you should see the created PVC:
+```
+# rbd list block --pool rbd
+pvc-c20495c0d5de11e8
+```
+
+# Additional features
+
+### Snapshots
+This example is based on [kubernetes-csi/external-snapshotter](https://github.com/kubernetes-csi/external-snapshotter), with a few tweaks to make it work along ceph-csi plugins. This is a basic example of the kubernetes snapshot feature. For more information and functionalities please refer to the [volume snapshot documentation](https://kubernetes.io/docs/concepts/storage/volume-snapshots/). 
+
+Make sure you enable the required [feature gate](https://kubernetes-csi.github.io/docs/Setup.html#csi-volume-snapshot-support) in your kubelet.
```

Comment:
in your kubelet -> in your Kubernetes cluster.

The feature gate applies to api server (for VolumeSnapshotClass api)